calibration/calibrate_model.sh

```diff
@@ -0,0 +1,285 @@
+#!/bin/bash
+###############################################################################
+# Copyright (C) 2024 Habana Labs, Ltd. an Intel Company
+###############################################################################
+
+set -e
+cd "$(dirname "$0")"
+
+ALLOWED_DEVICES=("g2" "g3")
+
+usage() {
+    echo
+    echo "Calibrate given MODEL_PATH for FP8 inference"
+    echo
+    echo "usage: ${0} <options>"
+    echo
+    echo "  -m    - [required] huggingface stub or local directory of the MODEL_PATH"
+    echo "  -d    - [required] path to source dataset (details in README)"
+    echo "  -o    - [required] path to output directory for fp8 measurements"
+    echo "  -b    - batch size to run the measurements at (default: 32)"
+    echo "  -l    - limit number of samples in calibration dataset"
+    echo "  -t    - tensor parallel size to run at (default: 1); NOTE: if t > 8 then we need a multi-node setup"
+    echo "  -r    - rank of unified measurements, it should be smaller than original rank number and should be a factor of the original rank number"
+    echo "  -u    - unify measurement results based on expert parallelism rules (default: False), expert parallelism unification rule is unique, card 1 expert measurement will be extended to card 0 if unified to x from 2x cards number"
+    echo "  -e    - set this flag to enable enforce_eager execution"
+    echo
+}
+
+cleanup_tmp() {
+	if [[ $(pwd) == *vllm-gaudi/calibration ]]; then
+		echo "Clearing temporary directory"
+		rm -rf nc_workspace
+		rm -rf inc_tmp
+	else
+		echo "Skipping temporary directory removal"
+	fi
+}
+
+create_measure_config() {
+    mkdir -p $1/$2/$3
+
+    model_name_lower=$(echo "$2" | tr '[:upper:]' '[:lower:]')
+
+    if [[ $model_name_lower =~ ^mixtral ]]; then
+        tmp_config="{\"method\": \"HOOKS\",\"mode\": \"MEASURE\",\"observer\": \"maxabs\",\"allowlist\": {\"types\": [], \"names\":  []},\"blocklist\": {\"types\": [], \"names\":  [\"self_attn\", \"lm_head\"]},\"quantize_weight\": false,\"dump_stats_path\": \"$1/$2/$3/inc_output\"}"
+    elif [[ $model_name_lower =~ ^deepseek ]]; then
```

Comment:
It would be great if could specify blockwise fp8 as an argument instead of hard-coding the model name as deepseek since there are other models like https://huggingface.co/Qwen/Qwen3-32B-FP8 which need a similar config.